challenge_scoring/metrics/scoring.py

```diff
@@ -141,101 +152,122 @@ def score_submission(streamlines_fname,
-    scores['algo_version'] = 5
-    scores['VC'] = VC
-    scores['IC'] = IC
-    scores['VCWP'] = VCWP
-    scores['NC'] = NC
-    scores['VB'] = nb_VB_found
-    scores['IB'] = nb_ib
-    scores['streamlines_per_bundle'] = streamlines_per_bundle
-    scores['total_streamlines_count'] = total_strl_count
-
-    # Get bundle overlap, overreach and f1-score for each bundle.
-    scores['overlap_per_bundle'] = {k: v["overlap"] for k, v in found_vbs_info.items()}
-    scores['overreach_per_bundle'] = {k: v["overreach"] for k, v in found_vbs_info.items()}
-    scores['overreach_norm_gt_per_bundle'] = {k: v["overreach_norm"] for k, v in found_vbs_info.items()}
-    scores['f1_score_per_bundle'] = {k: v["f1_score"] for k, v in found_vbs_info.items()}
+        save_valid_connections(found_vbs_info, sft, segmented_out_dir,
+                               segmented_base_name, out_tract_type,
+                               save_vbs=save_VBs,
+                               save_full_vc=save_full_vc)
+
+    candidate_ic_strl_indices = np.setdiff1d(range(total_strl_count),
+                                             VC_indices)
+    if compute_ic_ib:
+        logging.info("Starting IC, IB scoring")
+
+        candidate_ic_indices = []
+        rejected_indices = []
+
+        # Chosen from GT dataset
+        length_thres = 35.
+
+        # Filter streamlines that are too short, consider them as NC
+        for idx in candidate_ic_strl_indices:
+            if slength(sft.streamlines[idx]) >= length_thres:
+                candidate_ic_indices.append(idx)
+            else:
+                rejected_indices.append(idx)
+
+        logging.debug('Found {} candidate IC'
+                      .format(len(candidate_ic_indices)))
+        logging.debug('Found {} streamlines that were too short'
+                      .format(len(rejected_indices)))
+
+        ic_counts = 0
+        nb_ib = 0
+
+        if len(candidate_ic_indices):
+            additional_rejected_indices, ic_counts, nb_ib = \
+                group_and_assign_ibs(sft, candidate_ic_indices,  ROIs,
+                                     save_IBs, save_full_ic, segmented_out_dir,
+                                     segmented_base_name, ref_anat_fname,
+                                     out_tract_type)
+
+            rejected_indices.extend(additional_rejected_indices)
+
+        if ic_counts != len(candidate_ic_strl_indices) - len(rejected_indices):
+            raise ValueError("Some streamlines were not correctly assigned to "
+                             "NC")
+
+        IC = len(candidate_ic_strl_indices) - len(rejected_indices)
+    else:
+        IC = 0
+        nb_ib = 0
+        rejected_indices = candidate_ic_strl_indices
+
+    if len(rejected_indices) > 0 and save_full_nc:
+        out_nc_fname = os.path.join(
+            segmented_out_dir,
+            '{}_NC.{}'.format(segmented_base_name, out_tract_type))
+
+        save_tractogram(sft[rejected_indices], out_nc_fname)
```

Comment:
This fails for team 3_3. I had to add option bbox_valid_check=False. Should we add this here permanently?

Reply:
I would agree, especially if we want to keep supporting "old" tractograms like the original submissions.